12-Mar-2025 15:01:43.536 SEVERE [http-nio-8080-exec-4] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";


An error occurred at line: [23] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
f_name cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
l_name cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
email cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
phone_no cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
password cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [34] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
31:                 }
32: 
33:             } catch (Exception e){
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){


An error occurred at line: [35] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
32: 
33:             } catch (Exception e){
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();


An error occurred at line: [37] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();
39:                     conn.close();
40:                 }


An error occurred at line: [38] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();
39:                     conn.close();
40:                 }
41:             }


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";


An error occurred at line: [23] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
f_name cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
l_name cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
email cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
phone_no cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [24] in the jsp file: [/reg_connection.jsp]
password cannot be resolved to a variable
21: 
22:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
23:                 pstmt = conn.prepareParameter(sql);
24:                 pstmt.setString(f_name, l_name, email, phone_no, password);
25: 
26:                 int insertRow = pstmt.executeUpdate();
27:                 if (insertRow > 0){


An error occurred at line: [34] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
31:                 }
32: 
33:             } catch (Exception e){
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){


An error occurred at line: [35] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
32: 
33:             } catch (Exception e){
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();


An error occurred at line: [37] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
34:                 e.PrintStackTrace();
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();
39:                     conn.close();
40:                 }


An error occurred at line: [38] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
35:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
36:             } finally {
37:                 if(ptsmt != null && conn != null){
38:                     ptsmt.close();
39:                     conn.close();
40:                 }
41:             }


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:14:37.090 SEVERE [http-nio-8080-exec-8] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [21] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
18: 
19: 
20:             try{
21:                 Class.forName("com.mysql,cj.jdbc.Driver")
22:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
23: 
24:                 String f_name = request.getParameter("f_name");


An error occurred at line: [31] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
28:                 String password = request.getParameter("password");
29: 
30:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
31:                 pstmt = conn.prepareParameter(sql);
32:                 pstmt.setString(f_name, l_name, email, phone_no, password);
33: 
34:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [32] in the jsp file: [/reg_connection.jsp]
The method setString(int, String) in the type PreparedStatement is not applicable for the arguments (String, String, String, String, String)
29: 
30:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
31:                 pstmt = conn.prepareParameter(sql);
32:                 pstmt.setString(f_name, l_name, email, phone_no, password);
33: 
34:                 int insertRow = pstmt.executeUpdate();
35:                 if (insertRow > 0){


An error occurred at line: [42] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
39:                 }
40: 
41:             } catch (Exception e){
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){


An error occurred at line: [43] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
40: 
41:             } catch (Exception e){
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();
47:                     conn.close();
48:                 }


An error occurred at line: [46] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();
47:                     conn.close();
48:                 }
49:             }


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [21] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
18: 
19: 
20:             try{
21:                 Class.forName("com.mysql,cj.jdbc.Driver")
22:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
23: 
24:                 String f_name = request.getParameter("f_name");


An error occurred at line: [31] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
28:                 String password = request.getParameter("password");
29: 
30:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
31:                 pstmt = conn.prepareParameter(sql);
32:                 pstmt.setString(f_name, l_name, email, phone_no, password);
33: 
34:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [32] in the jsp file: [/reg_connection.jsp]
The method setString(int, String) in the type PreparedStatement is not applicable for the arguments (String, String, String, String, String)
29: 
30:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
31:                 pstmt = conn.prepareParameter(sql);
32:                 pstmt.setString(f_name, l_name, email, phone_no, password);
33: 
34:                 int insertRow = pstmt.executeUpdate();
35:                 if (insertRow > 0){


An error occurred at line: [42] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
39:                 }
40: 
41:             } catch (Exception e){
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){


An error occurred at line: [43] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
40: 
41:             } catch (Exception e){
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
42:                 e.PrintStackTrace();
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();
47:                     conn.close();
48:                 }


An error occurred at line: [46] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
43:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
44:             } finally {
45:                 if(ptsmt != null && conn != null){
46:                     ptsmt.close();
47:                     conn.close();
48:                 }
49:             }


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:24:59.936 SEVERE [http-nio-8080-exec-2] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(f_name, l_name, email, phone_no, password);
31: 
32:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [30] in the jsp file: [/reg_connection.jsp]
The method setString(int, String) in the type PreparedStatement is not applicable for the arguments (String, String, String, String, String)
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(f_name, l_name, email, phone_no, password);
31: 
32:                 int insertRow = pstmt.executeUpdate();
33:                 if (insertRow > 0){


An error occurred at line: [40] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
37:                 }
38: 
39:             } catch (Exception e){
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){


An error occurred at line: [41] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
38: 
39:             } catch (Exception e){
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();


An error occurred at line: [43] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();
45:                     conn.close();
46:                 }


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();
45:                     conn.close();
46:                 }
47:             }


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(f_name, l_name, email, phone_no, password);
31: 
32:                 int insertRow = pstmt.executeUpdate();


An error occurred at line: [30] in the jsp file: [/reg_connection.jsp]
The method setString(int, String) in the type PreparedStatement is not applicable for the arguments (String, String, String, String, String)
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(f_name, l_name, email, phone_no, password);
31: 
32:                 int insertRow = pstmt.executeUpdate();
33:                 if (insertRow > 0){


An error occurred at line: [40] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
37:                 }
38: 
39:             } catch (Exception e){
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){


An error occurred at line: [41] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
38: 
39:             } catch (Exception e){
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();


An error occurred at line: [43] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
40:                 e.PrintStackTrace();
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();
45:                     conn.close();
46:                 }


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
41:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
42:             } finally {
43:                 if(ptsmt != null && conn != null){
44:                     ptsmt.close();
45:                     conn.close();
46:                 }
47:             }


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:32:20.154 SEVERE [http-nio-8080-exec-8] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();
49:                     conn.close();
50:                 }


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();
49:                     conn.close();
50:                 }
51:             }


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql,cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();
49:                     conn.close();
50:                 }


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null && conn != null){
48:                     ptsmt.close();
49:                     conn.close();
50:                 }
51:             }


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:33:45.647 SEVERE [http-nio-8080-exec-1] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql.cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql.cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:36:20.502 SEVERE [http-nio-8080-exec-10] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql.cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [19] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete BlockStatements
16:             PreparedStatement pstmt = null;
17: 
18:             try{
19:                 Class.forName("com.mysql.cj.jdbc.Driver")
20:                 conn = DriverManager.getConnection(url,dbUsername,dbPassword);
21: 
22:                 String f_name = request.getParameter("f_name");


An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:38:45.507 SEVERE [http-nio-8080-exec-4] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [29] in the jsp file: [/reg_connection.jsp]
The method prepareParameter(String) is undefined for the type Connection
26:                 String password = request.getParameter("password");
27: 
28:                 String sql = "INSERT INTO user(f_name, l_name, email, phone_no, password) VALUES (?,?,?,?,?)";
29:                 pstmt = conn.prepareParameter(sql);
30:                 pstmt.setString(1, f_name);
31:                 pstmt.setString(2, l_name);
32:                 pstmt.setString(3, email);


An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:39:51.123 SEVERE [http-nio-8080-exec-2] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [44] in the jsp file: [/reg_connection.jsp]
The method PrintStackTrace() is undefined for the type Exception
41:                 }
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){


An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.PrintStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:40:26.482 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [Unable to compile class for JSP: 

An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.printStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.printStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [45] in the jsp file: [/reg_connection.jsp]
Syntax error, insert ";" to complete Statement
42: 
43:             } catch (Exception e){
44:                 e.printStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();


An error occurred at line: [47] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved to a variable
44:                 e.printStackTrace();
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){


An error occurred at line: [48] in the jsp file: [/reg_connection.jsp]
ptsmt cannot be resolved
45:                 out.println("<h3>Error message:"+ e.getMessage() +"</h3>")
46:             } finally {
47:                 if(ptsmt != null){
48:                     ptsmt.close();
49:                 }
50:                 if(conn != null){
51:                     conn.close();


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:72)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:192)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:524)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:373)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:345)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:331)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:396)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 15:40:29.710 SEVERE [http-nio-8080-exec-7] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/ServletDemo] threw exception [org.apache.jasper.JasperException: java.lang.ClassNotFoundException: org.apache.jsp.reg_005fconnection_jsp] with root cause
	java.lang.ClassNotFoundException: org.apache.jsp.reg_005fconnection_jsp
		at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:129)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:58)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:151)
		at org.apache.jasper.servlet.JspServletWrapper.getServlet(JspServletWrapper.java:189)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:410)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:377)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:325)
		at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 16:41:04.817 INFO [http-nio-8080-exec-9] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 16:41:04.817 SEVERE [http-nio-8080-exec-9] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: ServletDemo
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 16:41:54.617 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 16:41:54.617 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: ServletDemo
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 16:45:28.076 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 16:45:28.076 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: ServletDemo
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 16:46:18.172 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 16:46:18.172 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: ServletDemo
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 17:09:28.651 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 17:09:28.651 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: javax.servlet.http.HttpServlet
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at java.base/java.lang.ClassLoader.defineClass1(Native Method)
		at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1026)
		at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2354)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2223)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:813)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1339)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 17:10:23.087 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 17:10:23.087 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: ServletDemo
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
12-Mar-2025 17:24:24.013 INFO [http-nio-8080-exec-3] org.apache.catalina.core.ApplicationContext.log Marking servlet [ServletDemo] as unavailable
12-Mar-2025 17:24:24.014 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Allocate exception for servlet [ServletDemo]
	java.lang.ClassNotFoundException: javax.servlet.http.HttpServlet
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1378)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at java.base/java.lang.ClassLoader.defineClass1(Native Method)
		at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1026)
		at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2354)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClassInternal(WebappClassLoaderBase.java:2223)
		at org.apache.catalina.loader.WebappClassLoaderBase.findClass(WebappClassLoaderBase.java:813)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1339)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1187)
		at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:491)
		at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:473)
		at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:143)
		at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:758)
		at org.apache.catalina.core.StandardWrapper.allocate(StandardWrapper.java:575)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:115)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1743)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
		at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
		at java.base/java.lang.Thread.run(Thread.java:1575)
